Signal/src/ViewControllers/ConversationView/Cells/OWSMessageBubbleView.m

```diff
@@ -961,6 +1022,34 @@ - (CGSize)quotedMessageSize
     return CGSizeCeil(result);
 }
 
+- (CGSize)senderNameSizeWithBodyMediaSize:(CGSize)bodyMediaSize includeMargins:(BOOL)includeMargins
+{
+    OWSAssert(self.conversationStyle);
+    OWSAssert(self.conversationStyle.maxMessageWidth > 0);
+
+    if (!self.shouldShowSenderName) {
+        return CGSizeZero;
+    }
+
+    CGFloat hMargins = self.conversationStyle.textInsetHorizontal * 2;
+    const int maxTextWidth = (int)floor(self.conversationStyle.maxMessageWidth - hMargins);
+    [self configureSenderNameLabel];
+    CGSize result = CGSizeCeil([self.senderNameLabel sizeThatFits:CGSizeMake(maxTextWidth, CGFLOAT_MAX)]);
+
+    BOOL hasSeparateTextStackView = (self.isQuotedReply || bodyMediaSize.width > 0 || bodyMediaSize.height > 0);
```

Comment:
TIOLI: It's not obvious what bodyMediaSize has to do with whether or not there is a textStackView. Is that just a check to see if there **is** media? e.g. could we pull out a self.hasMedia getter for clarity?

Reply:
I completely agree, so I rewrote this in a subsequent PR.